tests/suites/test_suite_ssl.function

```diff
@@ -6094,3 +6113,81 @@ void ssl_ecjpake_set_password( int use_opaque_arg )
     USE_PSA_DONE( );
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_SSL_TLS_C*/
```

Comment:
Dependency can be defined in `.data` file or like here before the test function definition. Currently we have same dependency in both places. Not sure what is better solution for this case when test function is used only by one test case. I would suggest to leave it in `.data` file, because then test case information is in one place.

What do you think @mpg ?

If it should stay here:
```suggestion
/* BEGIN_CASE depends_on:MBEDTLS_SSL_TLS_C */
```

Reply:
Generally speaking there are 3 levels where we can declare dependencies:

1. Whole `.function` file, in the `BEGIN_DEPENDENCIES` block near the top of the file - when all functions in the file need this to build (compile & link).
2. Function definition in the `BEGIN_CASE` part - required when the function won't compile or link without this. Sometimes used also when it could build but could never succeed without this regardless of the test data (aka arguments): for example, the function uses SHA-256 and that's hardcoded, and the function will bail out before doing anything useful if SHA-256 isn't available, then it can declare a dependency on SHA-256 at that level.
3. Test case (in the `.data` file) - when the particular case can't do its job without with due to the actual value of the arguments.

Dependencies that are declared at one level don't need to be repeated at the levels below it. Here I think the function will not link if `MBEDTLS_SSL_TLS_C` so function declaration would be the right level, except actually we already have this dependency declared at the level of the whole file, so we don't need to repeat it here (even less in the `.data` file).

Thanks for pointing this out!

Reply:
This is something I found myself explaining more than once, so that suggests it should be documented in some place more permanent / discoverable than github comments on random PRs, but I never seem to find the time for such documentation updates :(

Reply:
This is documented in https://mbed-tls.readthedocs.io/en/latest/kb/development/test_suites